iris-mpc-cpu/src/execution/hawk_main.rs

```diff
@@ -155,38 +217,64 @@ pub struct HawkArgs {
     #[clap(flatten)]
     pub tls: Option<TlsConfig>,
 
+    /// Enables NUMA-aware optimizations.
     #[clap(long, default_value_t = false)]
     pub numa: bool,
 }
 
-/// HawkActor manages the state of the HNSW database and connections to other
-/// MPC nodes.
+/// Manages the state and execution of the HNSW-based MPC protocol (HAWK).
+///
+/// The `HawkActor` is the central component for the CPU-based matching engine. It orchestrates
+/// MPC sessions, manages the HNSW graphs and iris data stores for both eyes, and handles
+/// incoming requests. It is designed to be a long-lived object that holds
+/// the entire state of the MPC node.
+///
+/// # Responsibilities
+/// - **State Management:** Holds in-memory HNSW graphs (`graph_store`) and the underlying
+///   shared iris data (`iris_store`).
+/// - **Session Management:** Creates and manages `HawkSession`s, which provide the
+///   cryptographic context for MPC operations.
+/// - **Request Handling:** Processes `HawkRequest` batches.
+/// - **Persistence:** Coordinates with `GraphPg` to load the HNSW graph from and persist updates
+///   to a Postgres database.
+/// - **Anonymized Statistics:** Collects and aggregates distance data to generate anonymized
+///   statistics about match distributions.
 pub struct HawkActor {
+    /// Command-line arguments and configuration for the actor.
     args: HawkArgs,
 
-    // ---- Shared setup ----
+    // ---- Shared MPC & HNSW setup ----
+    /// The HNSW searcher, containing parameters and logic for graph traversal.
     searcher: Arc<HnswSearcher>,
+    /// A shared PRF key used to deterministically generate insertion layers for new nodes
+    /// in the HNSW graph, ensuring all parties build compatible graph structures.
+    /// Initialized once on startup.
     prf_key: Option<Arc<[u8; 16]>>,
 
-    // ---- My state ----
-    /// A size used by the startup loader.
+    // ---- Core State ----
+    /// The number of irises loaded into the database. Used during the initial load.
```

Comment:
Indeed, this counter exists only to support the caller using the "loader", conservatively implementing the given trait. Otherwise not used by HawkActor and not necessarily the actual size (not 100% sure whether that would work).